lib-blocking/caqti_blocking.ml

```diff
@@ -42,6 +42,21 @@ module System = struct
     let debug ?(src = default_log_src) = Logs.debug ~src
   end
 
+  module Stream = struct
+    type 'a t = 'a Seq.t
+
+    let from_fun f initial_state =
+      let state = ref initial_state in
+      let rec wrapped_f () =
+        match f !state with
+        | None -> Seq.Nil
+        | Some (a, new_state) ->
+          state := new_state;
+          Seq.Cons (a, wrapped_f)
+      in
+      wrapped_f
```

Comment:
This can be simplified to
```ocaml
    let rec from_fun f state () =
      (match f state with
       | None -> Seq.Nil
       | Some (a, state') -> Seq.Cons (a, from_fun f state'))
```